pkg/adapterManager/manager.go

```diff
@@ -95,32 +99,97 @@ func newCacheKey(kind aspect.Kind, cfg *config.Combined) (*cacheKey, error) {
 }
 
 // NewManager creates a new adapterManager.
-func NewManager(builders []adapter.RegisterFn, managers aspect.ManagerInventory, exp expr.Evaluator) *Manager {
+func NewManager(builders []adapter.RegisterFn, managers aspect.ManagerInventory, exp expr.Evaluator, sequential bool) *Manager {
 	mm, am := ProcessBindings(managers)
-	return newManager(newRegistry(builders), mm, exp, am)
+	return newManager(newRegistry(builders), mm, exp, am, sequential)
 }
 
-// newManager
-func newManager(r builderFinder, m map[aspect.Kind]aspect.Manager, exp expr.Evaluator, am map[aspect.APIMethod]config.AspectSet) *Manager {
+func newManager(r builderFinder, m map[aspect.Kind]aspect.Manager, exp expr.Evaluator, am map[aspect.APIMethod]config.AspectSet, sequential bool) *Manager {
 	return &Manager{
 		builders:    r,
 		managers:    m,
 		mapper:      exp,
 		methodMap:   am,
 		aspectCache: make(map[cacheKey]aspect.Wrapper),
+		sequential:  sequential,
 	}
 }
 
+// Shutdown gracefully drains the manager's worker pool.
+func (m *Manager) Shutdown() {
+	m.wg.Wait()
+}
+
 // Execute iterates over cfgs and performs the actions described by the combined config using the attribute bag on each config.
 func (m *Manager) Execute(ctx context.Context, cfgs []*config.Combined, attrs attribute.Bag, ma aspect.APIMethodArgs) aspect.Output {
-	// TODO: pool these arrays, we'll be making many and len(cfg) is constant for the life of the configuration.
-	results := make([]result, len(cfgs))
-	for i, cfg := range cfgs {
-		results[i] = result{cfg, m.execute(ctx, cfg, attrs, ma)}
+	numCfgs := len(cfgs)
+
+	// TODO: look into pooling both result array and channel, they're created per-request and are constant size for cfg lifetime.
+	results := make([]result, numCfgs)
+	resultChan := make(chan result, numCfgs)
+
+	if m.sequential {
+		for i, cfg := range cfgs {
+			results[i] = result{cfg, m.execute(ctx, cfg, attrs, ma)}
+		}
+	} else {
+		// schedule all the work that needs to happen
+		for _, cfg := range cfgs {
+			m.wg.Add(1)
+			pool.ScheduleWork(func() {
```

Comment:
This is a blocking call if the queue is full, the original uses a `select` to check for the context timing out while enqueuing to avoid enqueuing work for requests that've already timed out. We did this because once work is enqueued it cannot be removed. That behavior is a _bit_ harder to implement here. Before we were using the work queue `chan` directly so it was a simple:

		select {		
 		case <-ctx.Done():		
 			return aspect.Output{Status: status.WithDeadlineExceeded(fmt.Sprintf("failed to enqueue all config executions with err: %v", ctx.Err()))}		
 		case p.work <- task{ctx, cfg, attrs, ma, r}:		
 		}

Here the only we can do it is to check for the timeout then try to enqueue work otherwise. This means if a requests times out we'll still enqueue one adapter execution that's useless:

		select {		
 		case <-ctx.Done():		
 			return aspect.Output{Status: status.WithDeadlineExceeded(fmt.Sprintf("failed to enqueue all config executions with err: %v", ctx.Err()))}		
 		default:
                    pool.ScheduleWork(...)		
 		}